src/darwin/Framework/CHIP/MTRDeviceAttestationDelegateBridge.mm

```diff
@@ -43,18 +45,31 @@
                 NSError * error = (attestationResult == chip::Credentials::AttestationVerificationResult::kSuccess)
                     ? nil
                     : [MTRError errorForCHIPErrorCode:CHIP_ERROR_INTEGRITY_CHECK_FAILED];
-                [strongDelegate deviceAttestation:mDeviceController
-                               completedForDevice:device
-                            attestationDeviceInfo:deviceInfo
-                                            error:error];
+                if ([strongDelegate respondsToSelector:@selector
```

Comment:
Wonder if this should test not just for responding to this selector, but also `|| ![strongDelegate respondsToSelector:@selector(deviceAttestation:completedForDevice:attestationDeviceInfo:error:)]`, so that if the delegate fails to implement either, the exception complains about not responding to the new one instead of the deprecated one.

Reply:
I don't follow.....  The logic here is (intended to be) basically:
```
  if ([delegate respondsToSelector:A] || [delegate respondsToSelector:B]) {
    if ([delegate respondsToSelector:A]) {
      [delegate A];
    } else {
      [delegate B];
    }
  } else if (failed && ([delegate respondsToSelector:C] || [delegate respondsToSelector:D]) {
    if ([delegate respondsToSelector:C]) {
      [delegate C];
    } else {
      [delegate D];
    }
  }
```

There should be no exceptions here at all.

Reply:
You are right about that, please disregard!